recirq/qcqmc/trial_wf.py

```diff
@@ -0,0 +1,1584 @@
+                return energy.real, grad
+            else:
+                return 1e6, 1e6
+
+        if use_fast_gradients:
+            res = minimize(
+                fast_obj_grad, params, jac=True, method="BFGS", callback=progress_cb
+            )
+        else:
+            res = minimize(objective, params, callback=progress_cb)
+        if res.fun < best:
+            best = res.fun
+            best_res = res
+
+        if do_print:
+            print(res, flush=True)
+
+    assert best_res is not None
+    params = best_res.x
+    one_body_params = params[-n_one_body_params:]
+    two_body_params = params[:n_two_body_params]
+
+    wf, _ = get_evolved_wf(
+        one_body_params,
+        two_body_params,
+        initial_wf,
+        gate_generators,
+        n_orb,
+        restricted=restricted,
+        initial_orbital_rotation=initial_orbital_rotation,
+    )
+
+    one_body_cluster_mat = get_one_body_cluster_coef(
+        one_body_params, n_orb, restricted=restricted
+    )
+    # We need to change the ordering to match OpenFermion's abababab ordering
+    if not restricted:
+        index_rearrangement = np.asarray(
+            [i // 2 % (n_orb) + (i % 2) * n_orb for i in range(2 * n_orb)]
+        )
+        one_body_cluster_mat = one_body_cluster_mat[:, index_rearrangement]
+        one_body_cluster_mat = one_body_cluster_mat[index_rearrangement, :]
+
+    one_body_basis_change_mat = expm(one_body_cluster_mat)
+
+    if initial_orbital_rotation is not None:
+        if restricted:
+            one_body_basis_change_mat = (
+                initial_orbital_rotation @ one_body_basis_change_mat
+            )
+        else:
+            big_initial_orbital_rotation = np.zeros_like(one_body_basis_change_mat)
+
+            for i in range(len(initial_orbital_rotation)):
+                for j in range(len(initial_orbital_rotation)):
+                    big_initial_orbital_rotation[2 * i, 2 * j] = (
+                        initial_orbital_rotation[i, j]
+                    )
+                    big_initial_orbital_rotation[2 * i + 1, 2 * j + 1] = (
+                        initial_orbital_rotation[i, j]
+                    )
+
+            one_body_basis_change_mat = (
+                big_initial_orbital_rotation @ one_body_basis_change_mat
+            )
+
+    if do_print:
+        print("Hartree-Fock Energy:")
+        print(hf_energy)
+        initial_wf.print_wfn()
+        print("-" * 80)
+        print("FCI Energy:")
+        print(hamiltonian_data.e_fci)
+        print("-" * 80)
+        print(best_res)
+
+        print("-" * 80)
+        print("Ansatz Energy:")
+        print(np.real_if_close(wf.expectationValue(fqe_ham) + e_core))
+        wf.print_wfn()
+        print("Basis Rotation Matrix:")
+        print(one_body_basis_change_mat)
+        print("Two Body Rotation Parameters:")
+        print(two_body_params)
+
+    return one_body_params, two_body_params, one_body_basis_change_mat
```

Comment:
I agree.  I think this should be split up.

Reply:
Split everything into modules